### PR TITLE
docs: mention /icons in the template documentation

### LINKED ISSUE #8344: Document predefined emoji/icon images
When adding a workspace proxy, setting an icon for Coder app, etc you can set an icon URL. Right now, power users know to go to the template settings page to see the icon picker and get the right URL.

Can we add a `https://coder.example.com/icons` which just has the icon picker component so they can easily get an icon URL? It would also be nice if it statically rendered the paths for stuff like VS Code, JetBrains, and other icons we statically host. Then, in the `wsproxy create` help text and UI (and in the terraform provider docs) we can tell people to navigate there to see available icons.

### PR DESCRIPTION
**Pull request by aslilac**
Screenshot 2023-10-11 at 4 17 17 PM

Closes #8344

### CODE REVIEW

File: docs/templates/icons.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Icons
+
+---
+
+Coder uses icons in several places, including ones that can be configured
+throughout the app, or specified in your Terraform. They're specified by a URL,
+which can be to an image hosted on a CDN of your own, or one of the icons that
+come included with your Coder deployment.
+
+## Included icons
```
Comment (matifali) [inline docs/templates/icons.md:10]:
```suggestion
## Bundled icons
```

File: docs/templates/icons.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Icons
+
+---
+
+Coder uses icons in several places, including ones that can be configured
+throughout the app, or specified in your Terraform. They're specified by a URL,
+which can be to an image hosted on a CDN of your own, or one of the icons that
+come included with your Coder deployment.
```
Comment (matifali) [inline docs/templates/icons.md:8]:
```suggestion
In Coder, icons are utilized in various areas, and can be customized to enhance visual recognition and representation. They are defined via a URL, which can either be an image hosted on your CDN or one of the icons included with your Coder deployment.

1. **Template Icons**:
   - Customize icons to differentiate templates within the application.

2. **Terraform Resources**:
   - [`coder_parameter`](https://registry.terraform.io/providers/coder/coder/latest/docs/data-sources/parameter#icon): Specify an icon by providing a URL in the `icon` field.
   - [`coder_app`](https://registry.terraform.io/providers/coder/coder/latest/docs/resources/app#icon): Assign an icon by setting the `icon` field with a URL.
   - [`coder_script`](https://registry.terraform.io/providers/coder/coder/latest/docs/resources/script#icon): Define an icon by inserting a URL in the `icon` field.

3. **External Auth Providers**:
   - Set custom icons for [External Auth Providers](https://coder.com/docs/v2/latest/admin/external-auth) to visually distinguish between different authentication methods.

4. **Icon Sources**:
   - **CDNs**: Use any icon URL from a CDN accessible from your deployment. e.g., https://upload.wikimedia.org/wikipedia/commons/9/93/Amazon_Web_Services_Logo.svg
   - **Bundled Icons**: Use icons included with your Coder deployment by providing their path. e.g., "/icon/aws.svg"
```

**Review by matifali — COMMENTED** [review]
I think we should explain more on how an where we can use icons.

**Review by Kira-Pilot — APPROVED** [review]
I think @matifali's verbiage suggestions are great, but I consider this PR gtg after those changes are applied!

File: docs/templates/icons.md (hunk 2 above)
Comment (aslilac) [reply docs/templates/icons.md:8]: I didn't know all the places it was used, so this is helpful!

**Review by matifali — APPROVED** [review]
Looks great 👍

**Pull request merged**